aws/table_aws_sfn_state_machine_execution_history.go

```diff
@@ -346,20 +348,47 @@ func getRowDataForExecutionHistory(ctx context.Context, d *plugin.QueryData, arn
 		return nil, nil
 	}
 
-	params := &sfn.GetExecutionHistoryInput{
-		ExecutionArn: aws.String(arn),
+	maxLimit := int32(1000)
+	// If the requested number of items is less than the paging max limit
+	// set the limit to that instead
+	limit := d.QueryContext.Limit
+	if limit != nil {
+		if *limit < int64(maxLimit) {
+			maxLimit = int32(*limit)
+		}
 	}
 
 	var items []historyInfo
 
-	listHistory, err := svc.GetExecutionHistory(ctx, params)
```

Comment:
Does the AWS SDK offer any pagination with `svc.GetExecutionHistory`? Or was this table never iterating through pages before?

Reply:
This table did not implement the required pagination handling before: https://docs.aws.amazon.com/sdk-for-go/api/service/sfn/#SFN.GetExecutionHistory